File: scrimage-filters/src/main/scala/com/sksamuel/scrimage/filter/ShearFilter.scala
package com.sksamuel.scrimage.filter

import com.sksamuel.scrimage.BufferedOpFilter

class ShearFilter(xAngle: Double, yAngle: Double) extends BufferedOpFilter {
  val op = new thirdparty.jhlabs.image.ShearFilter()
  op.setXAngle(xAngle.toFloat)
  op.setYAngle(yAngle.toFloat)
  op.setResize(false)
}

object ShearFilter {
  def apply(xAngle: Double, yAngle: Double) = new ShearFilter(xAngle, yAngle)
}

File: scrimage-filters/src/main/java/com/sksamuel/scrimage/filter/ShearFilter.java
package com.sksamuel.scrimage.filter;

import java.awt.image.BufferedImageOp;

public class ShearFilter extends BufferedOpFilter {

    private final float xAngle;
    private final float yAngle;

    public ShearFilter(float xAngle, float yAngle) {
        this.xAngle = xAngle;
        this.yAngle = yAngle;
    }

    @Override
    public BufferedImageOp op() {
        thirdparty.jhlabs.image.ShearFilter op = new thirdparty.jhlabs.image.ShearFilter();
        op.setXAngle(xAngle);
        op.setYAngle(yAngle);
        op.setResize(false);
        return op;
    }
}
